src/main/java/com/importer/importer/DataSourceConfig.java

```diff
@@ -19,18 +18,35 @@ public class DataSourceConfig {
 
     @Value("${spring.datasource.username}")
     private String dataBaseUsername;
+    
+    @Value("${spring.datasource.password}")
+    private String dataBasePassword;
 
     @Autowired
-    private Environment environment;
+    private KMSUtil kmsUtil;
 
     @Bean
     public DataSource dataSource() throws IOException {
-        String dbPassword = environment.getProperty("spring.datasource.password");
+       // String dbPassword = environment.getProperty("spring.datasource.password");
+        boolean isEncrypted = checkDbPasswordIsEncrypted(dataBasePassword);
+        if(isEncrypted){
+          dataBasePassword = decryptEncryptedPassword(dataBasePassword);
```

Comment:
This is not the correct place to do the decryption of variables. 
As discussed today, create a config class which will handle the decryption of all the props.

Why did we do it in this way?